### ISSUE TITLE
"constructor" is read-only

### ISSUE THREAD

**Issue by golobitch** [MetaMask/snaps#1693]
I am building snap that communicates with XRPL. I searched through Endo and through this repo and I haven't found any pointers on how to solve this issue. The problem is in package `ripple-binary-codec`.

I narrowed it down and noticed that the breaking change is between versions 10.28 and 10.29.

I got following error in Flask version 10.29:
```
Error while running snap 'npm:@xrpl-snap/snap': "constructor" is read-only
```

What actually this error means, and how can I narrow this down further, to see what constructor is readonly? Error can be also visible if you try it yourself @ https://app.xrpl-snap.com

I can also give you access to source code since it is not yet public (it will be when we will launch it).

<img width="1800" alt="Screenshot 2023-08-27 at 16 56 57" src="https://github.com/MetaMask/snaps/assets/24848451/8381d1e9-98c5-4a32-b86c-d6c487c604fe">

**Comment by FrederikBolding** [comment]
@golobitch Hi! It looks like you are correct, `ripple-binary-codec` does seem to assign to the constructor at runtime. That is not allowed in the environment that snaps run under. See for example: https://unpkg.com/browse/ripple-binary-codec@1.5.0-beta.1/dist/types/hash.js

However, it does look like the latest version `1.9.0` may not have that problem. You can try that to see if that alleviates your problems!

**Comment by FrederikBolding** [comment]
@golobitch We made some changes recently that should help with this problem. I'll let you know once these changes land in the extension!

**Comment by golobitch** [comment]
@FrederikBolding no problem. I set up the local environment with SES and I got a nice stack trace. Issue was not in `ripple-binary-codec` but in underlying library (decimal.js). If someone will be also stucked, here is the issue: https://github.com/MikeMcl/decimal.js/blob/7f01abd83d5632005d5f97ddbea41e069ee69a74/decimal.js#L4894

I noticed that BigNumber.js does not have this problem, so I forked `ripple-binary-codec` and implemented BigNumber instead of Decimal, and now it is working.

Next step for us are tests & security audit :)

Thank you for your help, we can now close this issue.
